tests/test_job_management.py

```diff
@@ -2639,6 +2640,383 @@ def test_inspect_status(dispatcher_live_fixture, request_cred, roles, include_se
+            )
+
+    inspect_params = dict(
+                         token=encoded_token
+                     )
+    if pass_job_id:
+        inspect_params['job_id'] = job_id_done[:8]
+
+    c = requests.get(server + "/inspect-jobs",
+                     params=inspect_params)
+
+    if request_cred != 'private' or ('job manager' not in roles):
+        # email not supposed to be sent for public request
+        assert c.status_code == status_code
+        assert c.text == error_message
+    else:
+        jdata= c.json()
+        assert 'jobs' in jdata
+        assert type(jdata['jobs']) is list
+        if not pass_job_id:
+            assert len(jdata['jobs']) == 2
+            assert jdata['jobs'][0]['job_id'] == job_id_done or jdata['jobs'][1]['job_id'] == job_id_done
+            assert jdata['jobs'][0]['job_id'] == job_id_failed or jdata['jobs'][1]['job_id'] == job_id_failed
+        else:
+            assert len(jdata['jobs']) == 1
+            assert jdata['jobs'][0]['job_id'] == job_id_done
+
+        assert isinstance(jdata['jobs'][0]['job_status_data'], list)
+        if not pass_job_id:
+            assert isinstance(jdata['jobs'][1]['job_status_data'], list)
+
+        assert len(jdata['jobs'][0]['job_status_data']) == 1
+        if not pass_job_id:
+            assert len(jdata['jobs'][1]['job_status_data']) == 1
+
+        assert 'job_statuses' in jdata['jobs'][0]['job_status_data'][0]
+        assert isinstance(jdata['jobs'][0]['job_status_data'][0]['job_statuses'], list)
+        assert len(jdata['jobs'][0]['job_status_data'][0]['job_statuses']) == 1
+        if not pass_job_id:
+            assert 'job_statuses' in jdata['jobs'][1]['job_status_data'][0]
+            assert isinstance(jdata['jobs'][1]['job_status_data'][0]['job_statuses'], list)
+            assert len(jdata['jobs'][1]['job_status_data'][0]['job_statuses']) == 1
+
+        assert 'job_statuses_fn' in jdata['jobs'][0]['job_status_data'][0]
+        if not pass_job_id:
+            assert 'job_statuses_fn' in jdata['jobs'][1]['job_status_data'][0]
+            assert (jdata['jobs'][0]['job_status_data'][0]['job_statuses_fn'] ==
+                    f'scratch_sid_{session_id_done}_jid_{job_id_done}' or
+                    jdata['jobs'][0]['job_status_data'][0]['job_statuses_fn'] ==
+                    f'scratch_sid_{session_id_failed}_jid_{job_id_failed}')
+            assert (jdata['jobs'][1]['job_status_data'][0]['job_statuses_fn'] ==
+                    f'scratch_sid_{session_id_done}_jid_{job_id_done}' or
+                   jdata['jobs'][1]['job_status_data'][0]['job_statuses_fn'] ==
+                    f'scratch_sid_{session_id_failed}_jid_{job_id_failed}')
+        else:
+            assert jdata['jobs'][0]['job_status_data'][0]['job_statuses_fn'] == f'scratch_sid_{session_id_done}_jid_{job_id_done}'
+
+        assert jdata['jobs'][0]['job_status_data'][0]['job_statuses'][0]['job_status_file'] == 'job_monitor.json'
+        if not pass_job_id:
```

Comment:
this condition is repeated many times. Anyway, validation could be much more readable

Reply:
introducing the schema validation made allowed to remove many checks and made the whole more compact